### PR TITLE
VZ-5741 IngressTrait Controller support for Sticky Sessions in and out of Istio mesh

### PR DESCRIPTION
**Pull request by bryantidd**
# Description

This change updates the Ingress Trait Controller and Unit Tests to support sticky sessions both inside and outside an Istio mesh.
Fixes VZ-5471

# Checklist 

As the author of this PR, I have:

- [X] Checked that I included or updated copyright and license notices in all files that I altered
- [X] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: application-operator/controllers/ingresstrait/ingresstrait_controller.go [hunk 1]
```diff
@@ -699,9 +699,11 @@ func (r *Reconciler) createOrUpdateDestinationRule(ctx context.Context, trait *v
 				Namespace: trait.Namespace,
 				Name:      name},
 		}
+		namespace := &corev1.Namespace{}
+		_ = r.Client.Get(ctx, client.ObjectKey{Namespace: "", Name: trait.Namespace}, namespace)
```
Comment (pfmackin) [inline application-operator/controllers/ingresstrait/ingresstrait_controller.go:703]: need to check for error

File: application-operator/controllers/ingresstrait/ingresstrait_controller.go [hunk 2]
```diff
@@ -715,10 +717,23 @@ func (r *Reconciler) createOrUpdateDestinationRule(ctx context.Context, trait *v
...
+	istioEnabled := false
+	value, ok := namespace.Labels["istio-injection"]
+	if ok && value == "enabled" {
+		istioEnabled = true
+	}
 	destinationRule.Spec = istionet.DestinationRule{
-		Host: rule.Destination.Host,
+		Host: dest.Destination.Host,
 		TrafficPolicy: &istionet.TrafficPolicy{
+			Tls: &istionet.ClientTLSSettings{
```
Comment (pfmackin) [inline application-operator/controllers/ingresstrait/ingresstrait_controller.go:734]: I think we want to set ClientTLSSettings_DISABLE if the app is not in the mesh.

File: application-operator/controllers/ingresstrait/ingresstrait_controller.go [hunk 3]
```diff
@@ -733,6 +748,17 @@ func (r *Reconciler) mutateDestinationRule(destinationRule *istioclient.Destinat
 			},
 		},
 	}
+	if !istioEnabled {
```
Comment (pfmackin) [inline application-operator/controllers/ingresstrait/ingresstrait_controller.go:751]: I don't think we want port level settings, i think the TLS should just be disabled if the app is not in the mesh.  Also, the comment has Coherence in it.

File: application-operator/controllers/ingresstrait/ingresstrait_controller.go (hunk 1 above)
Comment (bryantidd) [reply application-operator/controllers/ingresstrait/ingresstrait_controller.go:703]: Will do

**Review by pfmackin — APPROVED** [review]

File: application-operator/controllers/ingresstrait/ingresstrait_controller.go [hunk 4]
```diff
@@ -715,10 +720,24 @@ func (r *Reconciler) createOrUpdateDestinationRule(ctx context.Context, trait *v
...
-func (r *Reconciler) mutateDestinationRule(destinationRule *istioclient.DestinationRule, trait *vzapi.IngressTrait, rule vzapi.IngressRule) error {
+// mutateDestinationRule changes the destination rule based upon a traits configuration
+func (r *Reconciler) mutateDestinationRule(destinationRule *istioclient.DestinationRule, trait *vzapi.IngressTrait, rule vzapi.IngressRule, services []*corev1.Service, namespace *corev1.Namespace) error {
+	dest, err := createDestinationFromRuleOrService(rule, services)
+	if err != nil {
+		return err
+	}
+
+	mode := istionet.ClientTLSSettings_DISABLE
+	value, ok := namespace.Labels["istio-injection"]
```
Comment (bobdonat) [inline application-operator/controllers/ingresstrait/ingresstrait_controller.go:731]: A namespace can istio-injection enabled but a pod could have it disabled. Coherence pods are an example of this.

**Pull request merged**